src/wp-includes/style-engine.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Style engine: Public functions
+ *
+ * This file contains a variety of public functions developers can use to interact with
+ * the Style Engine API.
+ *
+ * @package WordPress
+ * @subpackage StyleEngine
+ * @since 6.1.0
+ */
+
+
+/**
+ * Global public interface method to generate styles from a single style object, e.g.,
+ * the value of a block's attributes.style object or the top level styles in theme.json.
+ * See: https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-living/#styles and
+ * https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/
+ *
+ * Example usage:
+ *
+ * $styles = wp_style_engine_get_styles( array( 'color' => array( 'text' => '#cccccc' ) ) );
+ * // Returns `array( 'css' => 'color: #cccccc', 'declarations' => array( 'color' => '#cccccc' ), 'classnames' => 'has-color' )`.
+ *
+ * @access public
+ * @since 6.1.0
+ *
+ * @param array                 $block_styles The style object.
+ * @param array<string|boolean> $options      array(
+ *     'context' => (string|null) An identifier describing the origin of the style object, e.g., 'block-supports' or 'global-styles'. Default is 'block-supports'.
+ *                  When set, the style engine will attempt to store the CSS rules, where a selector is also passed.
+ *     'convert_vars_to_classnames' => (boolean) Whether to skip converting CSS var:? values to var( --wp--preset--* ) values. Default is `false`.
+ *     'selector'                   => (string) When a selector is passed, `generate()` will return a full CSS rule `$selector { ...rules }`, otherwise a concatenated string of properties and values.
+ * );.
```

Comment:
```suggestion
 * @param array $options {
 *     Optional. An array of options. Default empty array.
 *
 *     @type string|null $context                    An identifier describing the origin of the style object, such as 'block-supports' or 'global-styles'. 
 *                                                   When set, the style engine will attempt to store the CSS rules, where a selector is also passed.
 *                                                   Default 'block-supports'.
 *     @type bool        $convert_vars_to_classnames Whether to skip converting CSS var:? values to var( --wp--preset--* ) values. Default `false`.
 *     @type string      $selector                   When a selector is passed, generate() will return a full CSS rule `$selector { ...rules }`,
 *                                                   otherwise a concatenated string of properties and values.
 * }
```
This should also be done for all expanded `array` parameter documentation.


Additional Query: In "Whether to skip converting CSS var:? values", what's the `:?`?

Reply:
👍 

Good catch. It should be `var:preset|<PRESET_TYPE>|<PRESET_SLUG>`. I'll update all instances.